core/data.go

```diff
@@ -165,6 +169,67 @@ func (b Bundle) Size() uint64 {
 	return size
 }
 
+func (b Bundle) Serialize() ([]byte, error) {
+	if len(b) == 0 {
+		return []byte{}, nil
+	}
+	if len(b[0].Coeffs) == 0 {
+		return nil, errors.New("invalid bundle: the coeffs length is zero")
+	}
+	size := 0
+	for _, f := range b {
+		if len(f.Coeffs) != len(b[0].Coeffs) {
+			return nil, errors.New("invalid bundle: all chunks should have the same length")
+		}
+		size += bn254.SizeOfG1AffineCompressed + encoding.BYTES_PER_SYMBOL*len(f.Coeffs)
+	}
+	result := make([]byte, size+8)
+	buf := result
+	metadata := uint64(GnarkBundleEncodingFormat) | (uint64(len(b[0].Coeffs)) << 8)
+	binary.LittleEndian.PutUint64(buf, metadata)
+	buf = buf[8:]
+	for _, f := range b {
+		chunk, err := f.SerializeGnark()
+		if err != nil {
+			return nil, err
+		}
+		copy(buf, chunk)
+		buf = buf[len(chunk):]
+	}
+	return result, nil
+}
+
+func (b Bundle) Deserialize(data []byte) (Bundle, error) {
+	if len(data) < 8 {
+		return nil, errors.New("bundle data must have at least 8 bytes")
+	}
+	// Parse metadata
+	meta := binary.LittleEndian.Uint64(data)
```

Comment:
So there is only one serializatoin format based on bundle. The old format is based on chunk, I guess?

Reply:
That's right. We can put the Gob chunk into this bundle bytes as well, but since we'll move to Gnark which is strictly better (more compressed and much faster to encode/decode), then we don't need to bother old chunk any more.